backend/xivdpscalc/character/__init__.py

```diff
@@ -23,97 +25,142 @@ class Character:
     """
     The main class where damage is calculated. Initialized by providing a CharacterStatSpread.
     """
-    def __init__(self, job, stat_spread):
+    def __init__(self, job: Jobs, stat_spread: CharacterStatSpread):
+        """
+        :param job: Job enum, tells the character's job
+        :param stat_spread: dataclass containing all the attributes
+        """
         self.job = job
         self.wd = stat_spread.wd
-        self.stats = {}
-        self.stats[Stats.MAINSTAT] = Stat(Stats.MAINSTAT, stat_spread.mainstat)
-        self.stats[Stats.DET] = Stat(Stats.DET, stat_spread.det)
-        self.stats[Stats.CRIT] = ProbabalisticStat(Stats.CRIT, stat_spread.crit)
-        self.stats[Stats.DH] =  ProbabalisticStat(Stats.DH, stat_spread.dh)
-        self.stats[Stats.SPEED] = Stat(Stats.SPEED, stat_spread.speed)
-        self.stats[Stats.TEN] = Stat(Stats.TEN, stat_spread.ten)
-        self.stats[Stats.PIE] = Stat(Stats.PIE, stat_spread.pie)
-
-    def get_gcd(self):
+        self.character_stats = {
```

Comment:
Probably a

```python
Dict[Stats, Stat]
```

Would be appropriate here.

Reply:
i think this underscores the need to have names be a little more clear lol

is Stat just Stat Modifiers?

(also mypy is balking at this dict containing Stat but later a ProbablisticStat method is being called so I might separate them out)

Reply:
Ah, after looking at this, maybe:

```python
Dict[str, Stat]
```

and then use

```python
Stats.MAINSTAT.name: Stat(Stats.MAINSTAT, stat_spread_mainstat),
```

An enum's member has a `name` attribute which is just the stringified name of the member

Reply:
I think we should look at changing the names tho, but I don't know enough about the formula to feel confident making that call 

Though considering, originally it was `self.stats`, a dict containing `stats: stat` which is very confusing

Reply:
Might be best to undo the change and remove the types until the underlying logic can be revisted.

Reply:
Undo what change? You only suggested it lol

Reply:
The changes you made where you replaced all the `self.stats[Stats.FOO] = ...` with an inline dict declaration. Besides, that's not a typing change — that's a logic change.

Reply:
moved